### PR TITLE
Add missing tests for user.rb

### PR DESCRIPTION
**Pull request by shuheiktgw**
I just added some missing tests for `user.rb`!
~~Also, I refactored user.rb a bit by adding `new_user?` method.~~  I removed this commit!

Thank you! 👍

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -173,10 +173,10 @@ def confirm
   end
 
   def confirm!
-    new_user = !confirmed?
-
     skip_confirmation!
     save!
+
+    new_user = !confirmed?
```
Comment (Gargron) [inline app/models/user.rb]: Why does this change work, if skip already changes this value?

**Comment by Gargron** [comment]
@ykzts FYI I think rspec-retry does not work very cleanly, e.g. foreign key violations in 2nd and 3rd retries because it doesn't reset the database properly? This might confuse some contributors (the original try error is the real error)

**Review by Gargron — APPROVED** [review]

**Comment by ykzts** [comment]
dump.rdb?

**Comment by shuheiktgw** [comment]
Whoops, my apology. I'll create a PR to remove it.

**Pull request merged**